src/main/webapp/components/api/model-controller.service.js

```diff
@@ -36,22 +37,27 @@ angular.module('managementConsole.api')
             };
 
             /**
-             * Logs into the management endpoint and determines the launch type
+             * Logs into the management endpoint and determines the launch type.
+             * If username and/or password are not given, previously stored username and/or password are used.
              * @param {string} username - the username to use when connecting to the management endpoint
              * @param {string} password - the password to use when connecting to the management endpoint
              */
             ModelControllerClient.prototype.login = function (username, password) {
-                this.credentials.username = username;
-                this.credentials.password = password;
+                if (username !== undefined) { this.credentials.username = username; }
+                if (password !== undefined) { this.credentials.password = password; }
                 return this.readResource([], false, true).then(function (response) {
+                // TODO: Handle situation where response is received but login failed - that happens!
```

Comment:
Replicating the bug:

0. make sure that you are logged out of app
1. type http://localhost:3000/index.html#/clusters in url - app will open but with no name in upper right corner. That is because login got response, set authenticated to true but has no username really set. Here it should not get a response but fail instead, or it should be able to differentiate that response.
2. go to login screen, login - you willl see your username in upper right corner - that is how it should work.